plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Analyzer.java

```diff
@@ -254,29 +247,36 @@ public Void visitIf(final IfContext ctx) {
 
         final BlockContext blockctx0 = ctx.block(0);
         final StatementMetadata blocksmd0 = adapter.createStatementMetadata(blockctx0);
-        blocksmd0.last = ifsmd.last;
+        blocksmd0.lastSource = ifsmd.lastSource;
```

Comment:
After talking with you about this a little bit over Zoom, I'm going to leave the refactoring here for a different PR.  I think a lot of the structure for the metadata needs some work and more thought still.